modules/highgui/src/window_w32.cpp

```diff
@@ -596,6 +596,94 @@ void cvSetPropTopmost_W32(const char* name, const bool topmost)
     }
 }
 
+double cvGetPropVsync_W32(const char* name)
+{
+#ifndef HAVE_OPENGL
+    CV_UNUSED(name);
+    CV_Error(Error::OpenGlNotSupported, "Library was built without OpenGL support");
+    return -1;
+#else
+    CV_Assert(name);
+
+    if (!name)
+        CV_Error(Error::StsNullPtr, "NULL name string");
+
+    CvWindow* window = icvFindWindowByName(name);
+    if (!window)
+        CV_Error(Error::StsNullPtr, "NULL window");
+
+    // https://www.khronos.org/opengl/wiki/Swap_Interval
+    // https://www.khronos.org/registry/OpenGL/extensions/EXT/WGL_EXT_extensions_string.txt
+    // https://www.khronos.org/registry/OpenGL/extensions/EXT/WGL_EXT_swap_control.txt
+
+    if (!wglMakeCurrent(window->dc, window->hGLRC))
+        CV_Error(Error::OpenGlApiCallError, "Can't Activate The GL Rendering Context");
+
+    typedef const char* (APIENTRY* PFNWGLGETEXTENSIONSSTRINGEXTPROC)(void);
+    PFNWGLGETEXTENSIONSSTRINGEXTPROC wglGetExtensionsString = NULL;
+    wglGetExtensionsString = (PFNWGLGETEXTENSIONSSTRINGEXTPROC)wglGetProcAddress("wglGetExtensionsStringEXT");
+    if (wglGetExtensionsString == NULL)
+        CV_Error(Error::OpenGlApiCallError, "wglGetProcAddress failed to get wglGetExtensionsStringEXT");
```

Comment:
OpenGL support does not mean support of all required extensions. I propose not to thow exception, if the extension is not available in software or not implemented in GPU. It's better to return -1 as result of the function to user and it can handle it in safe way.

Reply:
I think that it's relevant for all calls to OpenGL extensions in "get" call.

Reply:
ok, I'll change them all to returning -1 in the get call